mtags/src/main/scala-2/scala/meta/internal/metals/MultilineStringFormattingProvider.scala

```diff
@@ -131,57 +159,110 @@ object MultilineStringFormattingProvider {
-    case (token: Constant.String, index: Int) =>
-      isMultilineString(sourceText, token) && hasStripMarginSuffix(
-        index,
-        tokens
-      ) && inToken(start, end, token)
+      andMultilineString: Boolean = true,
+      andHadStripMargin: Boolean = true,
+      andHasPipeInScope: Boolean = true
+  )(indent: => List[TextEdit]): PartialFunction[(Token, Int), List[TextEdit]] = {
+    case (token: Token.Constant.String, index)
+        if inToken(startPosition, endPosition, token)
+          && (if (andMultilineString) isMultilineString(token) else true)
+          && (if (andHadStripMargin) hasStripMarginSuffix(index, tokens)
+              else true)
+          && (if (andHasPipeInScope)
+                pipeInScope(
+                  startPosition,
+                  endPosition,
+                  sourceText,
+                  newlineAdded
+                )
+              else true) =>
+      indent
+    case (token: Interpolation.Start, index: Int)
+        if token.pos.start < startPosition.start && {
+          var endIndex = index + 1
+          while (!tokens(endIndex)
+              .isInstanceOf[Interpolation.End]) endIndex += 1
+          val endToken = tokens(endIndex)
+          endToken.pos.end > endPosition.end &&
+          (if (andMultilineString) isMultilineString(token) else true) &&
+          (if (andHadStripMargin) hasStripMarginSuffix(endIndex, tokens)
+           else true) &&
+          (if (andHasPipeInScope)
+             pipeInScope(
+               startPosition,
+               endPosition,
+               sourceText,
+               newlineAdded
+             )
+           else true)
+        } =>
+      indent
   }
-  private def shouldFormatInterpolationString(
-      sourceText: String,
-      tokens: Tokens,
+
+  private def indentTokensOnTypeFormatting(
       start: StartPosition,
-      endPosition: EndPosition
-  ): PartialFunction[(Token, Int), Boolean] = {
-    case (token: Interpolation.Start, index: Int)
-        if token.pos.start <= start.start => {
-      var endIndex = index + 1
-      while (!tokens(endIndex)
-          .isInstanceOf[Interpolation.End]) endIndex += 1
-      isMultilineString(sourceText, token) && hasStripMarginSuffix(
-        endIndex,
-        tokens
-      ) && tokens(endIndex).pos.end > endPosition.end
-    }
+      end: EndPosition,
+      tokens: Tokens,
+      newlineAdded: Boolean,
+      splitLines: Array[String],
+      position: Position,
+      text: String,
+      enableStripMargin: Boolean
+  ): List[TextEdit] = {
+    tokens.zipWithIndex
+      .collectFirst {
+        isStringOrInterpolation(start, end, text, newlineAdded, tokens)(
+          List(indent(splitLines, position))
+        ) orElse {
+          isStringOrInterpolation(
+            start,
+            end,
+            text,
+            newlineAdded,
+            tokens,
+            andHadStripMargin = false
+          ) {
+            if (enableStripMargin) {
```

Comment:
The issue is that we dont know whre the token ends ... for the string it works, but for the interpolation, we loose the information of the EndToken.. I dont know how to do that, except If  I search again with a while for endToken

Reply:
That's why I am thinking we should to that inside the case, where we have the end token. We can drop the partial function and just have a normal function returning a list of TextEdits and move searching for the end token from inside the case guard to the actual case. We return Nil if the end is not correct or `indent + stripMarginEdit(endToken.pos + 1)` otherwise